vep/README.md

```diff
@@ -0,0 +1,109 @@
+[Container Registry](https://cloud.google.com/container-registry/) of
+`my-project` on Google Cloud, se we can pick `[IMAGE_TAG]` as
+`gcr.io/my-project/vep_91`. Then push this image by:
+
+`gcloud docker -- push gcr.io/my-project/vep_91`
+
+**TODO**: Add `cloudbuild.yaml` files for both easy push and integration test.
+
+## How to download and package VEP databases
+
+Choose a local directory with enough space (e.g., ~20GB for homo_sapiens) to
+download and integrate different pieces of the VEP database or cache files.
+Then from within that directory run the
+[`build_vep_cache.sh`](build_vep_cache.sh) script. By default this script
+creates the database for human (homo_sapiens), referenec sequence `GRCh38`,
+and release 91 of VEP. These values can be overwritten by the following
+environment variables (note you should use the same VEP release
+that you used for creating VEP docker image above):
+
+* `VEP_SPECIES`
+* `GENOME_ASSEMBLY`
+* `ENSEMBL_RELEASE`
+
+## How to run VEP on GCP
+
+There is the helper script [`run_vep.sh`](run_vep.sh) that is added to the VEP
+docker image and can be used to run VEP. One way of running it on
+Google Cloud Platform (GCP) is through the [Pipelines API](
+https://cloud.google.com/genomics/v1alpha2/pipelines-api-command-line). For a
+sample `yaml` job description check
+[`sample_pipeline.yaml`](sample_pipeline.yaml).
+Here is a sample `gcloud` command that uses that file:
+
+```
+gcloud alpha genomics pipelines run \
+  --project my-project \
+  --pipeline-file sample_pipeline.yaml \
+  --logging gs://my_bucket/logs \
+  --inputs VCF_INFO_FILED=CSQ_RERUN
+```
+
+Note the `vep_cache_homo_sapiens_GRCh38_91.tar.gz` file that is referenced in
+the sample `yaml` file, is the output file that you get from the above database
+creation step.
+
+The [`run_vep.sh`](run_vep.sh) script relies on several environment variables
+that can be set to change the default behaviour. In the above example
+`VCF_INFO_FILED` is changed to `CSQ_RERUN` (the default is `CSQ_VT`).
+
+This is the full list of supported environment variables:
+
+* `SPECIES`: default is `homo_sapiens`
+* `GENOME_ASSEMBLY`: default is `GRCh38`
+* `NUM_FORKS`: The value to be set for
+[`--fork` option of VEP](
+http://.ensembl.org/info/docs/tools/vep/script/vep_options.html#opt_fork).
+default is 1.
+* `OTHER_VEP_OPTS`: Other options to be set for the VEP invocation, default is
+[`--everything`](
+http://ensembl.org/info/docs/tools/vep/script/vep_options.html#opt_everything)
+* `VCF_INFO_FILED`: The name of the info field to be used for annotations,
+default is `CSQ_VT`. See
+[`--vcf_info_field`](
+http://ensembl.org/info/docs/tools/vep/script/vep_options.html#opt_vcf_info_field)
+
+The following environment variables have to be set and point to valid storage
+locations:
+
+* `VEP_CACHE`: Where the tar.gz file, created in the above database creation
+step, is located.
+* `INPUT_FILE`: Note this can be either a VCF file or a compressed VCF file
```

Comment:
Hmm, `run_vep.sh` is a script that is run inside a docker container so there is a single instance of it running, what do you mean by running multiple instances? The reason I chose environment variables is (i) to simplify argument processing in the script and (ii) Pipelines API already uses environment variables for things like I/O files and to pass parameters so it makes sense to reuse them.

Reply:
To clarify: `run_vep.sh` is meant to be run as a single and only command through Pipelines API so it has a 1-1 relationship with containers. Please let me know if you feel this needs to be clarified further.

Reply:
Docker container doesn't stop one from running multiple instances of the same process. What I am suggesting is decouple run_vep.sh implementation from how it is used. I am not sure how well `--fork` works for VEP, but in case you want to utilize process parallelism, then you have to come back and change this script if implemented as is. Leave it to you.

Reply:
In principle, I agree with your comment that using command line arguments is preferred over environment variables but I was thinking of only the single script invocation scenario. The `--fork` option seems to be enough to benefit from local parallelization.

That said, as we discussed offline, this is probably going to change once we move things to the new forked repo. So I will update this runner script once those pieces are settled and might change this to use arguments too.